__tests__/pet.test.js

```diff
@@ -0,0 +1,218 @@
+const Pet = require('../src/pet');
+
+describe('constructor', () => {
+  let pet;
+  beforeEach(() => {
+   pet = new Pet('Jessie');
+  });
+    it('returns an object', () => {
+      expect(new Pet('Jessie')).toBeInstanceOf(Object);
+    });
+    it('sets the name property', () => {
+      expect(pet.name).toEqual('Jessie');
+    });
+    it('has an initial age of 0', () => {
+      expect(pet.age).toEqual(0);
+    });
+    it('has an initial hunger of 0', () => {
+     expect(pet.hunger).toEqual(0);
+    });
+    it('has initial fitness of 10', () => {
+      expect(pet.fitness).toEqual(10);
+    } )
+  });
+
+  describe('growUp', () => {
+    let pet;
+    beforeEach(() => {
+     pet = new Pet('Jessie');
+    });
+    it('increments the age by 1', () => {
+      pet.growUp();
+      expect(pet.age).toEqual(1);
+    });
+    it('increments the hunger by 5', () => {
+      pet.growUp();
+      expect(pet.hunger).toEqual(5);
+    });
+    it('decreases the fitness by 3', () => {
+      pet.growUp();
+      expect(pet.fitness).toEqual(7);
+    });
+    it('throws an error if the pet is not alive', () => {
+      pet.age = 30;
+      expect(() => pet.growUp()).toThrow('Your pet is no longer alive :(');
+    });
+  });
+
+  describe('walk', () => {
+    let pet;
+    beforeEach(() => {
+     pet = new Pet('Jessie');
+    });
+    it('increments the fitness by 4', () => {
+      pet.fitness = 4;
+      pet.walk();
+      expect(pet.fitness).toEqual(8);
+    });
+    it('increases fitness to a maximum of 10', () => {
+      pet.fitness = 8;
+      pet.walk();
+      expect(pet.fitness).toEqual(10);
+    });
+    it('throws an error if the pet is not alive', () => {
+      pet.age = 30;
+      expect(() => pet.walk()).toThrow('Your pet is no longer alive :(');
+    });
+  });
+
+  describe('feed', () => {
+    let pet;
+    beforeEach(() => {
+     pet = new Pet('Jessie');
+    });
+    it('decreases hunger by 3', () => {
+      pet.hunger = 5;
+      pet.feed();
+      expect(pet.hunger).toEqual(2);
+    });
+    it('decreases hunger to a minimum of 0', () => {
+      pet.hunger = 2;
+      pet.feed();
+      expect(pet.hunger).toEqual(0);
+    });
+    it('throws an error if the pet is not alive', () => {
+      pet.age = 30;
+      expect(() => pet.feed()).toThrow('Your pet is no longer alive :(');
+    });
+  });
+
+  describe('checkUp', () => {
```

Comment:
This a pretty solid suite of tests covering a lot of ground and functionality! There are a couple of things that could be done to make it more readable and prevent typos plus make it more maintainable:

1. Use constants for the various messages. These could be included in the `it` blocks using a template literal and directly in the `expect` argument, for example:
Define `walkMessage` at the top of the file so it will be in scope for all tests and does not need to be repeated for each suite: 
```
const walkMessage = "I need a walk";
```
Then for a test:
```
it(`returns '${walkMessage}' if pets fitness is 3 or less`, () => {
  pet.fitness = 3;
  pet.checkUp();
  expect(pet.checkUp()).toEqual(walkMessage)
});
```
This would mean less duplication, make things more maintainable and make the code neater to boot. (Note the use of single quotes around the actual message returned, this makes the text being returned in the test easier to separate from the rest of the message)
If all messages were stored like this the rest of these tests could be refactored in the same way.